src/react/reconcilation.js

```diff
@@ -283,10 +283,12 @@ export class Reconciler {
     try {
       this.realm.react.activeReconciler = this;
       let effects = this.realm.wrapInGlobalEnv(() =>
-        this.realm.evaluatePure(() =>
-          evaluateWithNestedParentEffects(this.realm, nestedEffects, () =>
-            this.realm.evaluateForEffects(resolveOptimizedClosure, /*state*/ null, `react nested optimized closure`)
-          )
+        this.realm.evaluatePure(
+          () =>
+            evaluateWithNestedParentEffects(this.realm, nestedEffects, () =>
+              this.realm.evaluateForEffects(resolveOptimizedClosure, /*state*/ null, `react nested optimized closure`)
+            ),
+          this._handleReportedSideEffect
```

Comment:
Why did Flow miss an argument here? Is it because it's marked as optional?

Can we mark it as explicit union of `null | (...) => ...` so we don't forget to pass it around in other places?

Reply:
We want to make it optional still as we're the only consumers of using the side-effect detection whilst pureScope is also used by the generic optimized function code path.

Reply:
I meant that the generic path can explicitly pass `null`. But if it’s not marked as optional argument then Flow should enforce that you either pass function or null, making the decision explicit.